src/lib/input/input-container.ts

```diff
@@ -257,11 +266,16 @@ export class MdInputContainer implements AfterContentInit {
   }
   private _hintLabel = '';
 
-  /** Text or the floating placeholder. */
+  /** Whether the placeholder should always float, never float or float as the user types. */
   @Input()
-  get floatingPlaceholder(): boolean { return this._floatingPlaceholder; }
-  set floatingPlaceholder(value) { this._floatingPlaceholder = coerceBooleanProperty(value); }
-  private _floatingPlaceholder: boolean = true;
+  get floatingPlaceholder() { return this._floatingPlaceholder; }
+  set floatingPlaceholder(value: MD_INPUT_PLACEHOLDER_TYPES) {
+    if (value && MD_INPUT_PLACEHOLDER_VALUES.indexOf(value) === -1) {
```

Comment:
I would be fine with just not validating this and treating anything other than 'always' and 'never' as auto, up to you though